src/res/heatmap.html

```diff
@@ -0,0 +1,4294 @@
+			if (level < 0 || level >= levels.length) {
+				continue;
+			}
+			const frames = findFramesBetween(levels[level], xFrom, xTo);
+			const x0 = xFrom;
+			for (let f of frames) {
+				const fw = f.width * s * px;
+				c.fillStyle = f.color;
+				c.fillRect((f.left - x0) * s * px, i * 16, fw, 15);
+				if (fw >= 21) {
+					const chars = Math.floor(fw / 7);
+					const title = f.title.length <= chars ? f.title : f.title.substring(0, chars - 2) + '..';
+					c.fillStyle = '#000000';
+					c.fillText(title, (f.left - x0) * s * px + 3, i * 16 + 12, fw - 6);
+				}
+			}
+		}
+		c.strokeRect(0, 7 * 16, w, 16);
+	}
+
+	canvas.onmousemove = function (e) {
+		previewX = e.offsetX;
+		previewY = e.offsetY;
+		if (e.shiftKey) {
+			showPreview();
+		} else {
+			preview_wrapper.style.display = 'none';
+		}
+
+		const h = Math.floor(e.offsetY / 16);
+		if (h >= 0 && h < levels.length) {
+			const f = findFrame(levels[h], e.offsetX / px + root.left);
+			if (f) {
+				hl.style.left = (Math.max(f.left - root.left, 0) * px + canvas.offsetLeft) + 'px';
+				hl.style.width = (Math.min(f.width, root.width) * px) + 'px';
+				hl.style.top = (h * 16 + canvas.offsetTop) + 'px';
+				hl.firstChild.textContent = f.title;
+				hl.style.display = 'block';
+				let details = f.details || '';
+				canvas.title = f.title + '\n(' + samples(f.width) + details + ', ' + pct(f.width, levels[0][0].width) + '%)';
+				canvas.style.cursor = 'pointer';
+				canvas.onclick = function (e) {
+					if (e.shiftKey) {
+						e.preventDefault();
+						searchForMethod(f.method);
+						canvas.onmousemove(e);
+						return;
+					}
+
+					if (e.ctrlKey) {
+						const n = e.shiftKey ? findPrevFame(levels[h], e.offsetX / px + root.left) : findNextFrame(levels[h], e.offsetX / px + root.left);
+						if (n) {
+							render(n, h);
+						}
+						canvas.onmousemove(e);
+						return;
+					}
+
+					if (f !== root) {
+						render(f, h);
+						canvas.onmousemove(e);
+					}
+				};
+				status.textContent = 'Function: ' + canvas.title;
+				status.style.display = 'inline';
+				return;
+			}
+		}
+		canvas.onmouseout();
+	}
+
+	canvas.onmouseout = function (e) {
+		hl.style.display = 'none';
+		status.textContent = '\xa0';
+		status.style.display = 'none';
+		canvas.title = '';
+		canvas.style.cursor = '';
+		canvas.onclick = '';
+		if (e !== undefined) {
+			preview_wrapper.style.display = 'none';
+			previewX = previewY = -1;
+		}
+	}
+
+	document.getElementById('reset').onclick = function () {
+		search(false);
+	}
+
+	class DataBuffer {
+		data;
+		pos = 0;
+
+		constructor(encodedData) {
+			this.data = encodedData;
+		}
+
+		nextVarInt() {
+			let res = 0;
+			let shift = 1;
+			let b;
+			do {
+				b = this.byteAt(this.pos++);
+				res += (b % 61) * shift;
+				shift *= 61;
+			} while (b >= 61);
+			return res;
+		}
+
+		nextInt6() {
+			return this.byteAt(this.pos++);
+		}
+
+		nextSymbol() {
+			return this.byteAt(this.pos++);
+		}
```

Comment:
This is a verbatim copy of `nextInt6`. Maybe, add `nextBase123` instead?

Reply:
Yeah, looks strange) Maybe somebody else took my body to code this peace)

Reply:
fixed